live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AncestorFilterApplierTest.java

```diff
@@ -63,38 +61,99 @@ class AncestorFilterApplierTest
     private AncestorFilterApplier objectUnderTest;
 
     @BeforeEach
-    void mockBasicData()
+    void mockNodeId()
     {
-        given(mockRepoEvent.getData()).willReturn(mockData);
-        given(mockData.getResource()).willReturn(mockResource);
-        given(mockFilter.path()).willReturn(mockPath);
+        given(mockResource.getId()).willReturn(NODE_ID);
     }
 
     @Test
-    void shouldNotFilterOutNullPrimaryHierarchyWhenEmptyAllowedAndEmptyDenied()
+    void givenEmptyAllowedAndEmptyDeniedFilters_whenNullPrimaryHierarchyOnCurrentNode_thenAllowNode()
     {
+        given(mockFilter.path()).willReturn(mockPath);
         given(mockResource.getPrimaryHierarchy()).willReturn(null);
         given(mockPath.allow()).willReturn(emptyList());
         given(mockPath.deny()).willReturn(emptyList());
 
         // when
-        boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
+        boolean result = objectUnderTest.allowNode(mockResource, mockFilter);
 
         // then
         assertTrue(result);
     }
 
     @Test
-    void shouldFilterOutWhenPrimaryHierarchyNullAndNonEmptyAllowedAndEmptyDenied()
+    void givenNonEmptyAllowedAndEmptyDeniedFilters_whenNullPrimaryHierarchyOnCurrentNode_thenDenyNode()
     {
+        given(mockFilter.path()).willReturn(mockPath);
         given(mockResource.getPrimaryHierarchy()).willReturn(null);
-        given(mockPath.allow()).willReturn(List.of(ANCESTOR));
+        given(mockPath.allow()).willReturn(List.of(ANCESTOR_ID));
         given(mockPath.deny()).willReturn(emptyList());
 
         // when
-        boolean result = objectUnderTest.applyFilter(mockRepoEvent, mockFilter);
+        boolean result = objectUnderTest.allowNode(mockResource, mockFilter);
 
         // then
         assertFalse(result);
     }
+
+    @Test
+    void whenNullPrimaryHierarchyOnPreviousNode_thenReturnCurrentlyAllowed()
+    {
+        given(mockResource.getPrimaryHierarchy()).willReturn(null);
+        final boolean currentlyAllowed = true;
+
+        // when
+        boolean resultForCurrentlyAllowed = objectUnderTest.allowNodeBefore(currentlyAllowed, mockResource, mockFilter);
+        boolean resultForCurrentlyDenied = objectUnderTest.allowNodeBefore(!currentlyAllowed, mockResource, mockFilter);
```

Comment:
I would split this into two tests.

Reply:
I didn't like that either but finally made use of `@ParametrizedTest` which suits here very well.